app_dart/lib/src/model/firestore/task.dart

```diff
@@ -0,0 +1,214 @@
+    statusInProgress,
+    statusNew,
+    statusSkipped,
+    statusSucceeded,
+  ];
+
+  static const List<String> finishedStatusValues = <String>[
+    statusCancelled,
+    statusFailed,
+    statusInfraFailure,
+    statusSkipped,
+    statusSucceeded,
+  ];
+
+  /// The timestamp (in milliseconds since the Epoch) that this task was
+  /// created.
+  ///
+  /// This is _not_ when the task first started running, as tasks start out in
+  /// the 'New' state until they've been picked up by an [Agent].
+  int? get createTimestamp => int.parse(fields!['createTimestamp']!.integerValue!);
+
+  /// The timestamp (in milliseconds since the Epoch) that this task started
+  /// running.
+  ///
+  /// Tasks may be run more than once. If this task has been run more than
+  /// once, this timestamp represents when the task was most recently started.
+  int? get startTimestamp => int.parse(fields!['startTimestamp']!.integerValue!);
+
+  /// The timestamp (in milliseconds since the Epoch) that this task last
+  /// finished running.
+  int? get endTimestamp => int.parse(fields!['endTimestamp']!.integerValue!);
+
+  /// The name of the task.
+  ///
+  /// This is a human-readable name, typically a test name (e.g.
+  /// "hello_world__memory").
+  String? get taskName => fields!['endTimestamp']!.stringValue!;
+
+  /// The number of attempts that have been made to run this task successfully.
+  ///
+  /// New tasks that have not yet been picked up by an [Agent] will have zero
+  /// attempts.
+  int? get attempts => int.parse(name!.split('_').last);
+
+  /// Whether this task has been marked flaky by .ci.yaml.
+  ///
+  /// See also:
+  ///
+  ///  * <https://github.com/flutter/flutter/blob/master/.ci.yaml>
+  ///
+  /// A flaky (`bringup: true`) task will not block the tree.
+  bool? get bringup => fields!['bringup']!.booleanValue!;
+
+  /// Whether the test execution of this task shows flake.
+  ///
+  /// Test runner supports rerun, and this flag tracks if a flake happens.
+  ///
+  /// See also:
+  ///  * <https://github.com/flutter/flutter/blob/master/dev/devicelab/lib/framework/runner.dart>
+  bool? get testFlaky => fields!['testFlaky']!.booleanValue!;
+
+  /// The build number of luci build: https://chromium.googlesource.com/infra/luci/luci-go/+/master/buildbucket/proto/build.proto#146
+  int? get buildNumber => fields!.containsKey('buildNumber') ? int.parse(fields!['buildNumber']!.integerValue!) : null;
+
+  /// The status of the task.
+  ///
+  /// Legal values and their meanings are defined in [legalStatusValues].
+  String get status {
+    final String taskStatus = fields!['status']!.stringValue!;
+    if (!legalStatusValues.contains(taskStatus)) {
+      throw ArgumentError('Invalid state: "$taskStatus"');
+    }
+    return taskStatus;
+  }
+
+  String setStatus(String value) {
+    if (!legalStatusValues.contains(value)) {
+      throw ArgumentError('Invalid state: "$value"');
+    }
+    fields!['status'] = Value(stringValue: value);
```

Comment:
This fields! everywhere, is this coming from the parent Document class?

Reply:
Yes. The `Task` is extending `Document`, where the `fields` is defined.

Reply:
Gotcha! Thanks for clarifying.